dotCMS/src/main/java/com/dotcms/content/elasticsearch/util/ESMappingUtilHelper.java

```diff
@@ -0,0 +1,329 @@
+package com.dotcms.content.elasticsearch.util;
+
+import com.dotcms.api.system.event.message.MessageSeverity;
+import com.dotcms.api.system.event.message.MessageType;
+import com.dotcms.api.system.event.message.SystemMessageEventUtil;
+import com.dotcms.api.system.event.message.builder.SystemMessageBuilder;
+import com.dotcms.content.elasticsearch.business.ESIndexAPI;
+import com.dotcms.content.elasticsearch.business.ESMappingAPIImpl;
+import com.dotcms.contenttype.business.ContentTypeAPI;
+import com.dotcms.contenttype.business.FieldFactory;
+import com.dotcms.contenttype.model.field.DataTypes;
+import com.dotcms.contenttype.model.field.DateField;
+import com.dotcms.contenttype.model.field.DateTimeField;
+import com.dotcms.contenttype.model.field.Field;
+import com.dotcms.contenttype.model.field.FieldVariable;
+import com.dotcms.contenttype.model.field.RadioField;
+import com.dotcms.contenttype.model.field.SelectField;
+import com.dotcms.contenttype.model.field.TextAreaField;
+import com.dotcms.contenttype.model.field.TextField;
+import com.dotcms.contenttype.model.field.TimeField;
+import com.dotcms.contenttype.model.field.WysiwygField;
+import com.dotcms.contenttype.model.type.ContentType;
+import com.dotcms.repackage.com.google.common.annotations.VisibleForTesting;
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.business.FactoryLocator;
+import com.dotmarketing.business.RelationshipAPI;
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.portlets.structure.model.Relationship;
+import com.dotmarketing.util.Logger;
+import com.dotmarketing.util.json.JSONObject;
+import com.jayway.jsonpath.JsonPath;
+import com.liferay.portal.language.LanguageException;
+import com.liferay.portal.language.LanguageUtil;
+import com.liferay.util.StringPool;
+import java.io.File;
+import java.io.IOException;
+import java.net.URL;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Locale;
+import java.util.Set;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+/**
+ * Helper class responsible of setting Elasticsearch mapping for content type fields
+ * @author nollymar
+ */
+public class ESMappingUtilHelper {
+
+    private static ContentTypeAPI contentTypeAPI;
```

Comment:
why static? they should be just final. Since it is a singleton you will keep just one reference anyhow

Reply:
The same for the methods, since the singleton is a self instance, methods should be instance not static

Reply:
Done